daemons/gptp/linux/src/daemon_cl.cpp

```diff
@@ -260,14 +260,39 @@ int main(int argc, char **argv)
 					return 0;
 				}
 			}
+			else if (strcmp(argv[i] + 1, "EP") == 0) {
+				if (i + 1 < argc) {
+					if (strcmp(argv[i + 1], "gm") == 0) {
+						ext_port_config.externalPortConfiguration = EXT_ENABLED;
+						ext_port_config.staticPortState = PTP_MASTER;
+					} else if (strcmp(argv[i + 1], "slave == 0")) {
+						ext_port_config.externalPortConfiguration = EXT_ENABLED;
+						ext_port_config.staticPortState = PTP_SLAVE;
+					} else {
+						printf("set external port configuration to be \"gm\" or \"slave\"\n");
+					}
+				} else {
+					printf("external port configuration needs to be specified"
+						    "to either \"gm\" or \"slave\"\n");
+				}
+			}
 			else if (strcmp(argv[i] + 1, "V") == 0) {
-				portInit.automotive_profile = true;
+				automotive_profile_config.automotiveProfile = true;
 			}
-			else if (strcmp(argv[i] + 1, "GM") == 0) {
-				portInit.isGM = true;
+			else if (strcmp(argv[i] + 1, "ETE") == 0) {
+				automotive_profile_config.automotiveTestMode = true;
 			}
-			else if (strcmp(argv[i] + 1, "E") == 0) {
-				portInit.testMode = true;
+			else if (strcmp(argv[i] + 1, "DC") == 0) {
+				automotive_profile_config.forceAsCapable = false;
+			}
+			else if (strcmp(argv[i] + 1, "DS") == 0) {
+				automotive_profile_config.automotiveState = false;
+			}
+			else if (strcmp(argv[i] + 1, "DR") == 0) {
+				automotive_profile_config.negotiateSyncRate = false;
+			}
+			else if (strcmp(argv[i] + 1, "ETX") == 0) {
+				automotive_profile_config.transmitAnnounce = true;
```

Comment:
What's the behavior if a user specifies one of these automotive options but doesn't specify the "enable automotive" option? From the looks of it, the variables all get silently cleared and gPTP will happily run in non-automotive mode. I think it would be worthwhile add a check here and do one of these options:
1) Notify the user that they need to pass in the "enable automotive" option and exit without running the protocol.
2) Alternatively, print a notice/info message and enable automotive.
It might be safer to assume they don't realize the option is for automotive only but at the same time they're using an automotive-only flag so the user might assume that the code is smart enough to realize that they want to use automotive. I'll leave it up to you. As long as the user gets something to make it clear what happened.

Reply:
I add notifications and make the program return if the user-defined configurations are not correct.
1. externalPortConfig needs to be enabled if automotive profile is enabled.
2. staticPortConfig should be configured either "gm" or "slave"
3. automotiveProfile needs to be enabled before configuring other automotive options.